nomenclature/definition.py

```diff
@@ -40,6 +53,33 @@ def __init__(self, path, dimensions=["region", "variable"]):
         if empty:
             raise ValueError(f"Empty codelist: {', '.join(empty)}")
 
+        # check that any variable region-processing-rename mappings are valid
+        if "variable" in self.dimensions:
+            items = [
```

Comment:
Mabye one more comment, I would put this variable validation code into its own function that is called in the `__init__`. Something like `_validate_variable` or along those lines. 
This should make it more obvious to the user that all these lines are responsible for variable validation